aptos-move/framework/aptos-framework/sources/PropertyList.move

```diff
@@ -0,0 +1,144 @@
+/// This Data Structure is designed to hold properties.
```

Comment:
This is nice. as discussed offline, I coded the propertylist based on the utility function I need for the Token before seeing this code. Hence this PropertyList has some functions that are not in the table.move


1. create a new PropertyList from keys and values, which can be from script function directly. It doesn't have to create tableItem and create a list from tableItem.

2. It also support get/update values for multiple keys that is used in token as well

3. I don't know if faster key find is possible since we can assume any ordering of the properties names.

In terms of naming, I don't want people to to use this PropertyList as a map. If they really want to, I used "list" explicitly in the name so developers know that they are using sth that is not a typical map.

Reply:
use the simpleMap for storing the data  now and add wrapper it with some utility function